src/main/java/com/yodle/vantage/component/dao/VersionDao.java

```diff
@@ -20,38 +20,21 @@
 import java.util.Map;
 import java.util.Optional;
 import java.util.Set;
-import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.jdbc.core.JdbcTemplate;
 import org.springframework.stereotype.Component;
 
-import com.google.common.collect.Sets;
 import com.yodle.vantage.component.domain.Dependency;
 import com.yodle.vantage.component.domain.Version;
+import com.yodle.vantage.component.domain.VersionId;
 
 @Component
 public class VersionDao {
-    public static final String UNKNOWN_VERSION = "unknown";
-    public static final String LATEST_VERSION = "latest";
-    public static final String UNDEFINED_VERSION = "undefined"; //Gradle will set versions to undefined if not set
-    public static final Set<String> RESERVED_VERSIONS = Sets.newHashSet(UNDEFINED_VERSION, UNKNOWN_VERSION, LATEST_VERSION);
     @Autowired private JdbcTemplate jdbcTemplate;
 
     public boolean createNewVersion(String componentName, String version) {
```

Comment:
I lifted the version string validation into ComponentService since we needed access to it anyway to determine what order we should create versions in.